src/store/datasets-layers.ts

```diff
@@ -0,0 +1,243 @@
+import { del, ref, set } from '@vue/composition-api';
+import vtkITKHelper from '@kitware/vtk.js/Common/DataModel/ITKHelper';
+import vtkImageData from '@kitware/vtk.js/Common/DataModel/ImageData';
+import vtkBoundingBox from '@kitware/vtk.js/Common/DataModel/BoundingBox';
+import { defineStore } from 'pinia';
+import { useImageStore } from '@/src/store/datasets-images';
+import { resample } from '../io/resample/resample';
+import { useDICOMStore } from './datasets-dicom';
+import {
+  DataSelection,
+  makeDICOMSelection,
+  makeImageSelection,
+  selectionEquals,
+  useImageID,
+} from './datasets';
+import { useErrorMessage } from '../composables/useErrorMessage';
+import { Manifest, StateFile } from '../io/state-file/schema';
+
+// differ from Image/Volume IDs with a branded type
+type DataSelectionKey = string & { __type: 'DataSelectionKey' };
+export type LayerID = string & { __type: 'LayerID' };
+
+export type Layer = {
+  selection: DataSelection;
+  id: LayerID;
+};
+
+const toDataSelectionKey = (selection: DataSelection) => {
+  const id =
+    selection.type === 'dicom' ? selection.volumeKey : selection.dataID;
+  return `${selection.type}::${id}` as DataSelectionKey;
+};
+
+const toDataSelectionFromKey = (key: DataSelectionKey) => {
+  const [type, id] = key.split('::');
+  if (type === 'dicom') return makeDICOMSelection(id);
+  if (type === 'image') return makeImageSelection(id);
+
+  throw new Error('Unknown DataSelection key');
+};
+
+const useImage = async (selection: DataSelection) => {
```

Comment:
Probably just call this `getImage`, since we use `use*` prefixes for composables. These functions may also be generally useful, so moving these out to another file may be a good idea. Thoughts?

Reply:
Yes.  Renamed to getImage and moved to `datasets.ts` to increase its dev visibility.